### PR TITLE
fix possible duplicate type

### PR DESCRIPTION
**Pull request by tcoopman**
@fertapric fixes: https://github.com/elixir-ecto/ecto/pull/2749#discussion_r226597676

### CODE REVIEW

**Comment by fertapric** [comment]
Thanks 💜!

I would add a test 🙂

**Comment by josevalim** [comment]
Thanks! Although I actually think the previous approach is fine. The type
in the front should be given higher priority in all cases and we document
that we cant return it. :)
-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

**Comment by fertapric** [comment]
> The type in the front should be given higher priority in all cases 

`Enum.into(errors, %{})` puts in the map the last `:type` value.

I agree this is a minor change but I think it's worthy if we can prevent some undesired side effects 🙂

File: test/ecto/changeset_test.exs [hunk 1]
```diff
@@ -262,6 +271,15 @@ defmodule Ecto.ChangesetTest do
     refute changeset.valid?
   end
 
+  test "cast/4: field has a custom invalid error message with a type parameter will be erased" do
```
Comment (fertapric) [inline test/ecto/changeset_test.exs]: Maybe `cast/4: ignores the :type parameter in custom errors`?

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**